shared/src/main/scala/crystal/ViewF.scala

```diff
@@ -4,18 +4,37 @@ import cats.implicits._
 import monocle.Lens
 import monocle.Optional
 import monocle.Prism
+import scala.concurrent.Promise
+import cats.effect.ContextShift
+import cats.effect.Async
+import cats.effect.Sync
 
 class ViewOptF[F[_], A](
   value:      Option[A],
-  val modOpt: (Option[A] => Option[A]) => F[Unit]
+  val modOpt: (Option[A] => Option[A]) => F[Unit] // Shouldn't actually be available in ViewF
 ) {
   @inline def getOption: Option[A] = value
 
   @inline def mod(f: A => A): F[Unit] = modOpt(_.map(f))
 
   @inline def set(a: A): F[Unit] = modOpt(_ => a.some)
 
-  @inline def setOpt(a: Option[A]): F[Unit] = modOpt(_ => a)
+  @inline def setOpt(a: Option[A]): F[Unit] =
+    modOpt(_ => a) // Shouldn't actually be available in ViewF
+
+  def modOptAndGetOption( // Shouldn't actually be available in ViewF
+    f:           Option[A] => Option[A]
+  )(implicit ev: Async[F], cs: ContextShift[F]): F[Option[A]] =
+    Sync[F].delay(Promise[Option[A]]).flatMap { p =>
```

Comment:
Isn't cat-effect `Deferred` essentially a `Promise`?

Reply:
That was my first attempt, but the problem is that a `Deferred` cannot be completed synchronously, whereas a `Promise` can.

I know this is not pure, but I can't find another way to extract the modified value from the bowels of the `mod` function.